### ISSUE TITLE
How to create a hierarchical softmax for resnet?

### ISSUE THREAD

**Issue by aezco** [keras-team/keras#12845]
I would like go implement  a hierarchical resnet architecture. However, I could not find any solution for this. For example, my data structure is like:

- class A
   - Subclass 1
   - Subclass 2
   - ....
- class B
   - subclass 6
   - ........

So i would like to predict the main class and then the subclass of the chosen/predicted mainclass. Can someone provide a simple example how to do this with generators?

Let say, we predict first the age group of a person (40,60), then the classifiers knows that the age is something between 40-60 years old and it will predict the apparent age 53. I cannot get the idea behind the implementation for this.
